pkg/plugin/plugin.go

```diff
@@ -367,6 +367,15 @@ func CmdAdd(args *skel.CmdArgs) error {
 		}
 		result = newResult
 		result.Interfaces = []*current.Interface{hostIface, result.Interfaces[0]}
+
+		for ifIndex, ifCfg := range result.Interfaces {
```

Comment:
This looks valid, but I believe we could fix the issue at its source, on line 359. Would you please investigate that option?

Reply:
Hello and thanks for the suggestion! Initially I attempted to fix at that point and did not understood why `newResult` had to be initialized there, but after reading the CNI spec and following the implementation logic, I thought like 359 is actually OK for these reasons:
- according to the CNI spec the IPAM returns no interfaces key: 

> Note that unlike regular CNI plugins, IPAM plugins should return an abbreviated Result structure that does not include the interfaces key, since IPAM plugins should be unaware of interfaces configured by their parent plugin except those specifically required for IPAM (eg, like the dhcp IPAM plugin).

- for this reason `newResult.Interfaces` is formed in order to be used with `ipam.ConfigureIface` and it actually only contains the container interface. Thus, line 359 correctly points to index 0.

So I chose to correct the index after IPAM is done at the end of ADD action.

Reply:
Fair enough.

Would you please extend the IPAM test to cover this too? To make sure the issue does not reappear.

Reply:
@phoracek Do we still have to return hostIface ? This is anyway not needed by metaplugin (example: multus) to build the pod network status. If not needed, we can remove this and it would fix the ip index problem.

Reply:
Good question. I don't know about any use for the host iface. OVS CNI only attaches to the bridge, never creates it, so I think it would be safe to drop it. It would certainly make things cleaner.

@gabidrg, would you please try what Peri proposed?

Reply:
As far as I see the veth pair is created by OVS CNI. The CNI spec mentions that host interfaces might be returned in ADD action result and I see use cases for having the host interface in result in other plugins (e.g. bridge) for instance to be used in the CHECK action (which should be implemented sooner or later for OVS CNI as well) or for handling of host side interface configuration (IP or VLAN tag?). Also, I am not sure about whether the SR-IOV use case might need it.

Reply:
I see. Then from my side, I'm just missing the test.

Reply:
AFAIK the metaplugin neither store the returned interfaces and nor pass it in the CmdCheck. 
The CmdCheck method still takes *skel.CmdArgs as argument as similar to CmdAdd, so its up to OVS CNI responsibility to find out host interface, so returning host interface in CmdAdd may not help here.
I'm not sure about meta-plugin uses bandwidth on the host interface since in this case it's just a OVS port attached with OVS bridge. is bandwidth still useful here ?

Reply:
To make sure we have the same representation of the issue, when I mentioned the result returning the interfaces, I meant the OVS-CNI plugin and its flow implementing ADD, DEL, CHECK actions and not Multus, as the metaplugin is just delegating the network configuration to OVS-CNI. In my understanding, when ADD is executed, the configuration it gets via *skel.CmdArgs triggers either the SR-IOV flow to configure a VF, or the general case when a veth pair is setup. In either cases the OVS bridge is already configured and not managed by OVS-CNI. Yet, it is OVS-CNI that manages the host side interface added to the bridge. As I mentioned above, I had two concerns about the result not containing the hostInterface:
- first in relation with its inner workings: subsequent actions such as CHECK might need it. This can be indeed solved as the DEL action does: it retrieves the hostInterface by using a query to ovsdb. So in this case, question is whether it is more efficient to keep the hostInterface in the result or not, in order to be used at some point by some CHECK action that has yet to be implemented.
- secondly in relation with chained plugins: I reported this issue because I found that sbr is not working while working for a telco use case. I also mentioned bandwitdth as other potential chained plugin that might use hostInterface, because it has use cases in telco QoS scenarios. The idea is to allow them to function properly if they need to operate over the host interface keeping in mind that they are unaware of the CNI implementation.

Reply:
ah! yes, I just noticed `prevResult` in `StdinData of CmdArgs` holds result obj which is then passed into check method. Then let's keep returning hostIface in the result.